browser/browsing_data/brave_clear_browsing_data_browsertest.cc

```diff
@@ -0,0 +1,389 @@
+    run_loop_.Run();
+    EXPECT_NE(removed_browser_, nullptr);
+    return CheckReturn();
+  }
+
+ private:
+  // BrowserListObserver
+  void OnBrowserRemoved(Browser* browser) override {
+    removed_browser_ = browser;
+    run_loop_.Quit();
+  }
+
+  size_t original_browsers_count_;
+  Browser* removed_browser_;
+  base::RunLoop run_loop_;
+
+  DISALLOW_COPY_AND_ASSIGN(BrowserRemovedObserver);
+};
+
+}  // namespace
+
+class BraveClearDataOnExitTest
+    : public InProcessBrowserTest,
+      public BraveClearBrowsingData::OnExitTestingCallback {
+ public:
+  BraveClearDataOnExitTest() = default;
+  void SetUpOnMainThread() override {
+    BraveClearBrowsingData::SetOnExitTestingCallback(this);
+  }
+
+  void TearDownOnMainThread() override {
+    // Borrowed from browser_browsertest.cc.
+    // Cycle the MessageLoop: one for each browser.
+    for (unsigned int i = 0; i < browsers_count_; ++i)
+      content::RunAllPendingInMessageLoop();
+
+    // Run the application event loop to completion, which will cycle the
+    // native MessagePump on all platforms.
+    base::ThreadTaskRunnerHandle::Get()->PostTask(
+        FROM_HERE, base::RunLoop::QuitCurrentWhenIdleClosureDeprecated());
+
+    base::RunLoop().Run();
+
+    // Take care of any remaining message loop work.
+    content::RunAllPendingInMessageLoop();
+
+    // At this point, quit should be for real now.
+    ASSERT_EQ(0u, chrome::GetTotalBrowserCount());
+  }
+
+  void TearDownInProcessBrowserTestFixture() override {
+    // Verify expected number of calls to remove browsing data.
+    EXPECT_EQ(remove_data_call_count_, expected_remove_data_call_count_);
+    BraveClearBrowsingData::SetOnExitTestingCallback(nullptr);
+  }
+
+  int remove_data_call_count() { return remove_data_call_count_; }
+
+  void SetExepectedRemoveDataCallCount(int count) {
+    expected_remove_data_call_count_ = count;
+  }
+
+  void SetExpectedRemoveDataRemovalMasks(int remove_mask, int origin_mask) {
+    expected_remove_mask_ = remove_mask;
+    expected_origin_mask_ = origin_mask;
+  }
+
+  void SetClearAll(PrefService* prefService) {
+    prefService->SetBoolean(browsing_data::prefs::kDeleteBrowsingHistoryOnExit,
+                            true);
+    prefService->SetBoolean(browsing_data::prefs::kDeleteDownloadHistoryOnExit,
+                            true);
+    prefService->SetBoolean(browsing_data::prefs::kDeleteCacheOnExit, true);
+    prefService->SetBoolean(browsing_data::prefs::kDeleteCookiesOnExit, true);
+    prefService->SetBoolean(browsing_data::prefs::kDeletePasswordsOnExit, true);
+    prefService->SetBoolean(browsing_data::prefs::kDeleteFormDataOnExit, true);
+    prefService->SetBoolean(browsing_data::prefs::kDeleteHostedAppsDataOnExit,
+                            true);
+    prefService->SetBoolean(browsing_data::prefs::kDeleteSiteSettingsOnExit,
+                            true);
+  }
+
+  int GetRemoveMaskAll() {
+    return ChromeBrowsingDataRemoverDelegate::DATA_TYPE_HISTORY |
```

Comment:
Should "Hosted App Data" be included here?

Reply:
"Hosted App Data" is defined as a combination of mask = `ChromeBrowsingDataRemoverDelegate::DATA_TYPE_SITE_DATA` and origin = `content::BrowsingDataRemover::ORIGIN_TYPE_PROTECTED_WEB`, so it should be already accounted for.